### PR TITLE
Leaderboard place

### LINKED ISSUE #596: [FEATURE]: Leaderboard positions
### Feature Summary

It would be great to have the position for each racer displayed on the leader board next to their username, reduces cognitive load and improves UX.

I will work on this if approved

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by BlakeLawrence**
---
title: Issue # | Added place column and position to leaderboard table
---

Discord Username: @BlakeLawrence 

## What type of PR is this? (select all that apply)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 🚧 Breaking Change
- [ ] 🧑‍💻 Code Refactor
- [ ] 📝 Documentation Update

## Description

added place column and position number to leaderboard

image

## Related Tickets & Documents

- Related Issue #596 
- Closes #596 

## Added/updated tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### CODE REVIEW

**Comment by bkbCodes** [comment]
Can we add Gold, Silver and Bronze crown for 1st, 2nd and 3rd rank respectively?

**Comment by BlakeLawrence** [comment]
> We can add Gold, Silver and Bronze crown for 1st, 2nd and 3rd rank respectively

That's a great idea,  once this PR gets merged, feel free to go ahead on that

File: packages/app/src/app/leaderboard/users-table.tsx [hunk 1]
```diff
@@ -25,8 +25,27 @@ export function UsersTable({
   data: UserWithResults[];
   pageCount: number;
 }) {
+  // Calculates the "place" property for each user based on "averageCpm"
+  const sortedData = [...data].sort(
+    (a, b) => Number(a.averageCpm) - Number(b.averageCpm),
+  );
+
+  sortedData.forEach((user, index) => {
+    user.place = index + 1;
```
Comment (scape76) [inline packages/app/src/app/leaderboard/users-table.tsx:34]: So, on each page there is going to be places from 1 - per_page  only? We need to consider the current page and multiply by it so we get the global ranking, not just from the page

Comment (BlakeLawrence) [reply packages/app/src/app/leaderboard/users-table.tsx:34]: @scape76 apologies for the delay, I have only had an opportunity to read this now. Ah ok, I didnt quite notice how off I was with my solution.

I will have a think about this and get around to it as soon as possible, although I have really struggled while not seeing anyone else but myself in the leaderboard - any suggestions?

Comment (BlakeLawrence) [reply packages/app/src/app/leaderboard/users-table.tsx:34]: @bkbCodes I did like your suggestion about the gold and silver etc - do you by any chance want to take the leaderboard positions and include your stuff while you are at it? id be happy to hand over as I am really pressed for time

Comment (bkbCodes) [reply packages/app/src/app/leaderboard/users-table.tsx:34]: Sure, I gladly accept your offer

Comment (scape76) [reply packages/app/src/app/leaderboard/users-table.tsx:34]: > 
propably you didnt race enough

Comment (scape76) [reply packages/app/src/app/leaderboard/users-table.tsx:34]: > @bkbCodes I did like your suggestion about the gold and silver etc - do you by any chance want to take the leaderboard positions and include your stuff while you are at it? id be happy to hand over as I am really pressed for time
propably you didnt race enough

**Pull request closed without merging**